design/ironic-interaction/ironic-background-docs.md

```diff
@@ -0,0 +1,346 @@
+on semi-dedicated "provisioning" network should be leveraged. This network
+may be reused by other services and systems, but can also be leveraged in
+the case of hardware discovery. The most important factor is that this
+network does not have an external DHCP server attached.
+
+Some specific hardware types in ironic do support use of virtual media
+to boot the deployment ramdisk, however this funcationality is not
+available with vendor neutral IPMI. Newer protocols such as Redfish
+may support virtual media, but not as of the time of this document
+having been composed. Use of virtual media is thus should be considered
+out-of-scope. Other methods such as booting directly from iSCSI should
+also be considered out-of-scope in this use case.
+
+## How ironic writes an operating sysem image to baremetal?
+
+Ironic supports two fundimental types of disk images.
+
+Whole-disk and Partition (or filesystem) images.
+
+Due to the nature of partition images, where a bootloader and configuration
+must be put in place and configured, it is simpler to focus on the use of
+whole disk images.
+
+The basic workflow consists of:
+
+1. Booting the deployment ramdisk
+2. The deployment ramdisk checks in with the ironic introspection
+   service which updates configuration information stored about the
+   baremetal machine.
+3. The deployment ramdisk checks in with the ironic, and leverages a
+   MAC address to help identify the physical machine in the hardware
+   inventory.
+4. Ironic initiates deployment by first identifying the root disk upon
+   which the disk image is to be written. By default this will be the
+   smallest storage device available, and can be overridden via explict
+   configuration set into ironic.
+5. The deployment ramdisk downloads the image to be written and steams
+   that to the storage device.
+6. If defined as part of the deployment, ironic will add an additional
+   partition for a configuration drive. Ironic will then write the
+   configuraiton drive to disk, and reboot the machine.
+
+There are some additional steps that ironic performs, mainly fixing
+partition table data with GPT based partition tables in order to
+prevent issues after deployment, but these steps are incorporated
+as part of the deployment sequence to help ensure that the machine
+will deploy successfully without issues.
+
+## What connectivity is required?
+
+To load the baremetal hardware:
+
+* DHCP
+* TFTP (if iPXE is not natively supported by the network interfaces.)
+* HTTP in order to download kernel/ramdisk images via HTTP over a TCP
+  socket connection.
+
+The discovery and deployment ramdisk image needs to be able to:
+
+* DHCP and DNS
+* Connect to the ironic inspector API endpoint, which operates
+  on port 5050/TCP by default.
+* Connect to the ironic API endpoint, which operates on port
+  6385/TCP by default.
+* The deployment ramdisk will likely need to be able to reach an
```

Comment:
The iscsi deployment interface, but we've put extensive work to offer the files via the conductor if all else fails to help drive forward the use of the direct deployment interface because it is less sensitive to transient packet loss.

Reply:
That feature still requires access to an HTTP(s) endpoint hosted somewhere, even if it is just where ironic is being hosted. This becomes a little conflated because with the direct deploy interface, any url can be defined.